File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/services/SupabaseManager.kt
package nl.bsodetheepot.mobile.data.services

import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.postgrest.Postgrest
import io.github.jan.supabase.realtime.Realtime
import io.github.jan.supabase.storage.Storage

/** Centrale Supabase-client, analoog aan lib/supabase.ts in de webapp. */
object SupabaseManager {
    val client = createSupabaseClient(
        supabaseUrl = Secrets.SUPABASE_URL,
        supabaseKey = Secrets.SUPABASE_ANON_KEY,
    ) {
        install(Auth)
        install(Postgrest)
        install(Realtime)
        install(Storage)
    }
}

File: TheepotMobile/TheepotMobile/Services/SupabaseManager.swift
import Foundation
import Supabase

/// Centrale Supabase-client, analoog aan lib/supabase.ts in de webapp.
enum SupabaseManager {
    static let client = SupabaseClient(
        supabaseURL: Secrets.supabaseURL,
        supabaseKey: Secrets.supabaseAnonKey
    )
}
